visualization/app/codeCharta/ui/copyToClipboardButton/util/clipboardString.spec.ts

```diff
@@ -0,0 +1,35 @@
+import { buildTextOfFiles } from "./clipboardString"
+import { FileToValue } from "./getFilenamesWithHighestMetrics"
+
+describe("buildTextOfFiles", () => {
+	it("should return only header line if there is only one attribute without files", () => {
+		const result = buildTextOfFiles(WITH_ONE_FILELESS_ATTRIBUTE)
+
+		expect(result).toBe("FUNCTIONS\n")
+	})
+	it("should return valid string if there two attributes", () => {
+		const result = buildTextOfFiles(WITH_TWO_ATTRIBUTES)
+
+		expect(result).toBe(`RLOC\n` + `\t• fileA (12)\n` + `COMMENTS\n` + `\t• fileA (14)\n`)
+	})
+	it("should return valid string if there is one attribute with many files", () => {
+		const result = buildTextOfFiles(WITH_ONE_ATTRIBUTE)
+
+		expect(result).toBe(`MCC\n` + `\t• file1 (100)\n` + `\t• file2 (84)\n` + `\t• file3 (122)\n`)
+	})
+})
+const WITH_ONE_ATTRIBUTE = new Map<string, FileToValue[]>([
+	[
+		"mcc",
+		[
+			{ name: "file1", value: 100 },
+			{ name: "file2", value: 84 },
+			{ name: "file3", value: 122 }
+		]
+	]
+])
+const WITH_TWO_ATTRIBUTES = new Map<string, FileToValue[]>([
+	["rloc", [{ name: "fileA", value: 12 }]],
+	["comments", [{ name: "fileA", value: 14 }]]
+])
+const WITH_ONE_FILELESS_ATTRIBUTE = new Map<string, FileToValue[]>([["functions", []]])
```

Comment:
I prefer to have the constants at the beginning of the file. This would also preserve consistency between test files.

Reply:
Actually, it is different in some test files. For example, deltaGenerator.spec.ts .

In deltaGenerator, the reason was the following: When you enter a test file, then you do not want your eyes to immediately focus on the details of your mock objects. You do not care about these objects before you have seen the basic structure of the relevant test.

You want to see the tests. Otherwise, people start hiding their mock objects in mockData.ts again which Cedric wanted to prevent. 

### Mock objects vs config objects

But we all agree that constant config-objects have their place at the top of the file. This is, because we often search for them explicitly! Sometimes, a whole ticket basically says "change config property value". But this does not apply to test objects.